### PR TITLE
apiv2 binding test fixes

### PR DESCRIPTION
**Pull request by baude**
a recent refactor in the bindings broke the tests.  quick fixes to get them working again.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: pkg/bindings/connection.go [hunk 1]
```diff
@@ -113,13 +113,12 @@ func (c *Connection) DoRequest(httpBody io.Reader, httpMethod, endpoint string,
 		err      error
 		response *http.Response
 	)
-	safePathValues := make([]interface{}, len(pathValues))
+	var safePathValues []interface{}
```
Comment (jwhonce) [inline pkg/bindings/connection.go]: If you keep make(), then use for loop iteration value, eg:

safePathValue[i] = url.QueryEscape(pv)

Then you don't need to append.  Shaves some time off.

File: pkg/bindings/connection.go [hunk 2]
```diff
@@ -113,13 +113,12 @@ func (c *Connection) DoRequest(httpBody io.Reader, httpMethod, endpoint string,
...
 	// Make sure path values are http url safe
 	for _, pv := range pathValues {
 		safePathValues = append(safePathValues, url.QueryEscape(pv))
 	}
 	safeEndpoint := fmt.Sprintf(endpoint, safePathValues...)
```
Comment (jwhonce) [inline pkg/bindings/connection.go]: Please add a TODO about converting to url library.  Thanks.

File: pkg/bindings/connection.go [hunk 3]
```diff
@@ -150,8 +149,8 @@ func GetConnectionFromContext(ctx context.Context) (*Connection, error) {
 	if c == nil {
 		return nil, errors.New("unable to get connection from context")
 	}
-	conn := c.(Connection)
-	return &conn, nil
+	conn := c.(*Connection)
```
Comment (jwhonce) [inline pkg/bindings/connection.go]: Why not just return the pointer?

Comment (baude) [reply pkg/bindings/connection.go]: it actually fails to work work like that. it was on the casting side.

File: pkg/bindings/connection.go (hunk 2 above)
Comment (baude) [reply pkg/bindings/connection.go]: sure

**Comment by baude** [comment]
updated with review comments

**Pull request merged**